tokio/src/sync/mpsc/chan.rs

```diff
@@ -41,9 +41,13 @@ pub(crate) trait Semaphore {
 
     fn add_permit(&self);
 
+    fn add_permits(&self, n: usize);
+
     fn close(&self);
 
     fn is_closed(&self) -> bool;
+
+    fn num_acquired(&self) -> usize;
```

Comment:
What does this function do? The name is unclear to me.

Reply:
Colloquially, it is a count of yet-to-be-received successfully sent messages.  The Semaphore has the property that `is_idle` is true if this function returns 0.  I see what you mean.  Do you find `num_used` or `used_permits`[ as proposed here](https://stackoverflow.com/a/29453095) more clear?

Reply:
How about `num_permits_in_use`?